src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -0,0 +1,209 @@
+    public static final double INCHES_TO_CM = -1;
+    public static final double GEAR_RATIO = -1;
+    public static final double DISTANCE_PER_PULSE = WHEEL_DIAMETER_IN_INCHES * INCHES_TO_CM * GEAR_RATIO * Math.PI;
+
+    public static final double TRACK_WIDTH = -1;
+
+    public static final int SECONDS_IN_MINUTE = 60;
+
+    private static Drivetrain instance;
+
+    private final AHRS gyro;
+
+    private final DifferentialDriveOdometry odometry;
+    private final DifferentialDriveKinematics kinematics;
+    private final RamseteController ramseteController;
+    private final Field2d field2d;
+
+    private final RelativeEncoder leftEncoder;
+    private final RelativeEncoder rightEncoder;
+
+    private final Namespace trapezoidSettingsNamespace = namespace.addChild("trapezoid profile settings");
+    private final Supplier<Double> trapezoidVelocity = trapezoidSettingsNamespace.addConstantDouble("velocity", 0);
+    private final Supplier<Double> trapezoidAcceleration = trapezoidSettingsNamespace.addConstantDouble
+            ("acceleration", 0);
+    private final TrapezoidProfileSettings trapezoidProfileSettings;
+
+    private final Namespace leftPIDNamespace = namespace.addChild("left pid");
+    private final Supplier<Double> kPLeft = leftPIDNamespace.addConstantDouble("kP", 0);
+    private final Supplier<Double> kILeft = leftPIDNamespace.addConstantDouble("kI", 0);
+    private final Supplier<Double> kDLeft = leftPIDNamespace.addConstantDouble("kD", 0);
+    private final Supplier<Double> waitTimeLeft = leftPIDNamespace.addConstantDouble("wait time", 0);
+    private final Supplier<Double> toleranceLeft = leftPIDNamespace.addConstantDouble("tolerance", 0);
+    private final PIDSettings leftPIDSettings;
+
+    private final Namespace rightPIDNamespace = namespace.addChild("right pid");
+    private final Supplier<Double> kPRight = rightPIDNamespace.addConstantDouble("kP", 0);
+    private final Supplier<Double> kIRight = rightPIDNamespace.addConstantDouble("kI", 0);
+    private final Supplier<Double> kDRight = rightPIDNamespace.addConstantDouble("kD", 0);
+    private final Supplier<Double> waitTimeRight = rightPIDNamespace.addConstantDouble("wait time", 0);
+    private final Supplier<Double> toleranceRight = rightPIDNamespace.addConstantDouble("tolerance", 0);
+    private final PIDSettings rightPIDSettings;
+
+    private final Namespace feedForwardNamespace = namespace.addChild("feed forward");
+    private final Supplier<Double> kS = feedForwardNamespace.addConstantDouble("kS", 0);
+    private final Supplier<Double> kV = feedForwardNamespace.addConstantDouble("kV", 0);
+    private final Supplier<Double> kA = feedForwardNamespace.addConstantDouble("kA", 0);
+    private final Supplier<Double> kG = feedForwardNamespace.addConstantDouble("kG", 0);
+    private final FeedForwardSettings feedForwardSettings;
+
+    private final Namespace anglePIDNamespace = namespace.addChild("angle pid");
```

Comment:
Move this whole set of fields for the angle PID to be below the left and right PID